### DISCUSSION TITLE
A problem of loading the data of the set(knee_singlecoil_val)

### DISCUSSION THREAD

**Discussion by kukudehang** [facebookresearch/fastMRI#195] [Q&A] [answered]
Hello, when I use mri_data.SlicaDataset to load dataset(knee_singlecoil_val) , then I use fftc.ifft2c_new function to transform the data. Why the value of the data is close to 0.

**Comment by mmuckley** [comment, marked as answer]
Converting to a discussion Q&A as this does not document a bug.

Hello @kukudehang, did you look at the resulting image? The values may be tiny, but for non-parametric MRI we usually don't care about the units so much, just the relative values. The values are close to 0 because of how the original data was converted to floating point precision.

**Reply by kukudehang** [reply]
Thanks for your answer. I have look at the resulting image, but  something may  be wrong in the way i use the code.
The code is as follow: First, dataset = mri_data.SliceDataset(root= ~, challenge='singlecoil' , transform=(a function transform the data to tensor)).The root just contain one h5 file from the knee_singlecoil_val.So dataset is a four dimension tensor(number_of_slice, Height, Width, 2(complex)). Second, I use torch.view_as_complex(fastmri.fftc.ifft2c_new(dataset[i])) to get the image_slice i (complex). Third,  I use the code np.abs(image_slice[i].numpy()) to get the amplitude abs_imge_slice i. Finally, i use the code  PIL.Image.fromarray(abs_image_slice[i]) to see the resulting image. But the resulting image are black. 
Is there anything wrong in the way i use the code? Should I do a normalization to get the relative values and then visualize the resulting image?

**Reply by mmuckley** [reply]
Hello @kukudehang, you may need to do some normalizations. You can see the exact code we use for processing in the [`UNetDataTransform`](https://github.com/facebookresearch/fastMRI/blob/13560d2f198cc72f06e01675e9ecee509ce5639a/fastmri/data/transforms.py#L281-L388) - if you copy the single-coil options you should get an image.

One other thing to check would be which slice you're looking at - you don't want to look at an edge slice which might actually have no anatomy in it.
